src/__tests__/restLink.ts

```diff
@@ -3512,22 +3815,27 @@ describe('Mutation', () => {
         }
       `;
 
-      await toPromise<Result>(
-        execute(link, {
-          operationName: 'publishPost',
-          query: createPostMutation,
-          variables: { input: post },
-        }),
+      await firstValueFrom<Result>(
+        execute(
+          link,
+          {
+            query: createPostMutation,
+            variables: { input: post },
+          },
+          { client: dummyClient },
+        ),
```

Comment:
By the way, it might help to create a helper that will prepopulate the `client`  for you. See our [link helpers](https://github.com/apollographql/apollo-client/blob/d0b8a7ef10a6551e90e8bf5e6b705a30453b39ad/src/testing/internal/link.ts#L14) for inspiration. We import that and [alias to `execute`](https://github.com/apollographql/apollo-client/blob/d0b8a7ef10a6551e90e8bf5e6b705a30453b39ad/src/link/error/__tests__/index.ts#L15) to make these calls a bit nicer.

We should probably publish that as an official test utility, but until then, that might be helpful 🙂